check-services.php

```diff
@@ -61,10 +66,68 @@
     }
 }
 
-$sql = 'SELECT * FROM `devices` AS D, `services` AS S WHERE S.device_id = D.device_id ' . $where . ' ORDER by D.device_id DESC';
+if (!isset($query)) {
+    $query = "SELECT * FROM `devices` WHERE `disabled` = 0 $where ORDER BY `device_id` ASC";
+}
+
+foreach (dbFetch($query) as $device) {
+    $sql = 'SELECT * FROM `devices` AS D
+ INNER JOIN `services` AS S ON S.device_id = D.device_id AND D.device_id = '.$device['device_id'].'
+ LEFT JOIN `devices_attribs` as A ON D.device_id = A.device_id AND A.attrib_type = "override_icmp_disable"
+ ORDER by D.device_id DESC;';
+
+    foreach (dbFetchRows($sql) as $service) {
+    // Run the polling function if the associated device is up, "Disable ICMP Test" option is not enabled,
+    // or service hostname/ip is different from associated device
+        if ($service['status'] === "1" || ($service['status'] === '0' && $service['status_reason'] === 'snmp') ||
+        $service['attrib_value'] === 'true' || ($service['service_ip'] !== $service['hostname'] &&
+        $service['service_ip'] !== $service['ip'] )) {
```

Comment:
`$service['ip']` is the device IP here so you will need to run that through `inet6_ntop()` first.

Reply:
Done.